streaming/src/main/scala/org/apache/spark/streaming/scheduler/JobScheduler.scala

```diff
@@ -67,6 +67,8 @@ class JobScheduler(val ssc: StreamingContext) extends Logging {
     eventLoop.start()
 
     listenerBus.start(ssc.sparkContext)
+    // Estimators receive updates from batch completion
+    ssc.graph.getInputStreams.map(_.rateController).foreach(ssc.addStreamingListener(_))
```

Comment:
This probably should be done before the listener bus is started. Otherwise it may miss some events.